components/region_cache_memory_engine/src/background.rs

```diff
@@ -360,35 +390,49 @@ impl Runnable for BackgroundRunner {
                 }
             }
             BackgroundTask::LoadRange => {
+                let mem_usage = self.core.memory_controller.mem_usage();
+                if mem_usage
+                    > (self.core.memory_controller.soft_limit_threshold()
```

Comment:
maybe we can stop loading when it just hits the soft limits.

Reply:
sounds reasonable.